### PR TITLE
Sofias Weather App, Week 4 Web Dev Bootcamp spring 2024

### PR DESCRIPTION
**Pull request by sofia32057**
## Netlify link
[https://sofias-weather-app.netlify.app/](https://sofias-weather-app.netlify.app/)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0" />
+    <title>Sofias Weather App | Week 4</title>
+    <link
```
Comment (HIPPIEKICK) [inline index.html:9]: Nice!

File: script.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+};
+
+// Get min temp for entire day from api data
+const getMinTemp = (day, data) => {
+  const date = convertTime(day.dt, data.timezone).getDate();
+  const min = data.list
+    .filter(entry => convertTime(entry.dt, data.timezone).getDate() === date)
+    .sort((a, b) => a.main.temp_min - b.main.temp_min)[0];
+  return Math.floor(min.main.temp_min);
+};
```
Comment (HIPPIEKICK) [inline script.js:87]: Nice solution!

File: script.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+  const noons = getNoons(forecastData);
+  weatherForecast.innerHTML = "";
+  noons.forEach(obj => {
+    const maxTemp = getMaxTemp(obj, forecastData);
+    const minTemp = getMinTemp(obj, forecastData);
+    const day = setWeekday(obj.dt_txt);
+    weatherForecast.innerHTML += `
+      <div class="forecast-day">
+        <p class="forecast-day-label">${day}</p>
+        <i class="weather-icon"><img src="https://openweathermap.org/img/wn/${obj.weather[0].icon}.png"/></i>
```
Comment (HIPPIEKICK) [inline script.js:140]: Seems to be working with the day/night icons 👍

File: script.js [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+  return new Promise((resolve, reject) => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(resolve, reject, {
+        timeout: 5000,
+      });
+    } else {
+      throw new Error("No geolocation available");
+    }
+  });
+};
```
Comment (HIPPIEKICK) [inline script.js:236]: Beautiful!

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Beautifully executed Sofia! You've accomplished a lot this week; you followed the design closely and implemented extra feature's that still fits in well - like animations, the hamburger, the search bar etc.

One thing I would like you to fix is the responsiveness, as your content doesn't fit on a 320px screen:
- Make your app responsive (it should look good on devices from 320px width up to 1600px)

Skärmavbild 2024-03-04 kl  12 20 16

File: script.js [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+const getNoons = forecastData =>
+  forecastData.list.filter(obj => obj.dt_txt.includes("12:00"));
+
+// Convert milliseconds to readable time HH:MM, converted to local time
+const convertTime = (seconds, timezone) => {
+  const local = seconds + timezone;
+  return new Date(local * 1000);
+};
+
+// Construct minutes and hours with two digits UTC
```
Comment (Citronskal) [inline script.js:64]: Impressive that you remember to convert the time to 2 digits! Took me a while to realize

File: script.js (hunk 4 above)
Comment (Citronskal) [reply script.js:236]: Second to this, it's very thoughtful and user-friendly to build this function. it shows that you treated it like a real application not just thinking about copying the design (which i did !)

File: style.css [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+  --hover-color: #00000005;
+  --dark-color: #b64124;
+  --day-gradient: linear-gradient(90deg, #8589ff 0%, #e8e9ff 100%);
+  --night-gradient: linear-gradient(270deg, #6264a2 50.98%, #222350 51.14%);
+  --font-size: 20px;
+  --space: 30px;
+  --radus: 50vw 15vw;
+  --border-radius: 2.5rem;
+  --box-shadow: 0px 4px 8px 0px #00000059;
+}
```
Comment (Citronskal) [inline style.css:16]: I am very impressed you defined CSS custom properties first to make the code clean.

File: script.js [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+};
+
+// -- Event listeners
+menuBtn.addEventListener("click", () => toggleHide(navWrapper));
+menuClose.addEventListener("click", () => toggleHide(navWrapper));
+
+scrollArrow.addEventListener("click", () => {
+  toggleFullscreen(body);
+  toggleHide(weatherForecast);
+});
```
Comment (Citronskal) [inline script.js:258]: I loveee your animation, perhaps make the whole screen scrolls down as the forecast unfold will be nice?

scrollArrow.addEventListener("click", () => {
  toggleFullscreen(body);
  toggleHide(weatherForecast);
  window.scrollTo({
    top: window.innerHeight, 
    behavior: 'smooth' 
  });
});

**Review by Citronskal — COMMENTED** [review]
Hi Sofia, I am extremely impressed at your work! Your app is high-fidelity and user-friendly. Your codes are clean and easy to read. I learned so much for reviewing your codes. Keep up the amazing work!

File: script.js (hunk 5 above)
Comment (sofia32057) [reply script.js:64]: Me too!

File: style.css (hunk 6 above)
Comment (sofia32057) [reply style.css:16]: I love it, makes it so much easier!

File: script.js (hunk 7 above)
Comment (sofia32057) [reply script.js:258]: YES! Great idea!

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**